### PR TITLE
fix(loader): translate logging level to Python scale in EnvDictionary…

### LINKED ISSUE #3203: [Bug]: Latest WIP console window up on startup without verbose logging
### ✈ Pre-Flight checks

- [x] I **don't have SentinelOne** antivirus installed (see above for the solution)
- [x] I have **searched in the issues** (open and closed) but couldn't find a similar issue
- [x] I have **searched in the [pyRevit Forum](https://discourse.pyrevitlabs.io)** for similar issues
- [x] I already **followed the [installation troubleshooting guide](https://pyrevitlabs.notion.site/Installation-issues-f6495adb56254e9e8021f8a68e3b2ab7) thoroughly**
- [x] I am using the **latest pyRevit Version**

### 🐞 Describe the bug

I get this on every startup with the latest WIP, even if the logging level is set quite low:

Image

I suspect some logging not being set to the correct level.

### ⌨ Error/Debug Message

```shell
see above
```

### ♻️ To Reproduce

_No response_

### ⏲️ Expected behavior

_No response_

### 🖥️ Hardware and Software Setup (please complete the following information)

```shell
6.1.0.26047+2349
```

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by tay0thman**
…Seeder (#3203)

# fix(loader): translate logging level to Python scale in EnvDictionarySeeder 

## Problem

The console window (ScriptConsole) appears on every Revit startup showing DEBUG-level messages, even when the user's logging level is set to Quiet (the default). This was introduced in commit 47a34c3f when EnvDictionarySeeder was added for #3108..

## Root Cause

EnvDictionarySeeder.Seed() writes PYREVIT_LOGGINGLEVEL using PyRevitConfig.LoggingLevel, which returns a pyRevit enum (0=Quiet, 1=Verbose, 2=Debug). But the Python logger reads this value and compares it against Python logging module levels (10=DEBUG, 20=INFO, 30=WARNING).
When Quiet mode is active, the seeder writes 0. The Python logger evaluates record.levelno >= 0 — always True — so every DEBUG message passes through to ScriptIO, which calls output.Show(), making the console visible.

## Fix

Add a ToPythonLoggingLevel() translation method that maps:

0 (Quiet) → 30 (logging.WARNING)
1 (Verbose) → 20 (logging.INFO)
2 (Debug) → 10 (logging.DEBUG

## Checklist

 Default (Quiet) → no console window flash on startup
 Verbose mode → console shows INFO+ messages
 Debug mode → console shows DEBUG+ messages
 All pyRevit buttons still function after fix
 Reload pyRevit → console behavior correct
 Legacy loader path (new_loader=False) unaffected

---

## Related Issues

If applicable, link the issues resolved by this pull request:

- Resolves #3203 

Thank you for contributing to pyRevit! 🎉

### CODE REVIEW

**Comment by tay0thman** [comment]
@jmcouffin , Tested on Revit 2023,2024 and 2026 (IP 2.7), I welcome you and Copilot's Review.

File: dev/pyRevitLoader/pyRevitAssemblyBuilder/SessionManager/EnvDictionarySeeder.cs [hunk 1]
```diff
@@ -45,6 +45,21 @@ internal static class EnvDictionarySeeder
...
+        /// See: pyrevitlib/pyrevit/coreutils/logger.py DEFAULT_LOGGING_LEVEL
+        /// </summary>
+        private static int TranslateLoggingLevel(int pyrevitLevel)
+        {
+            switch (pyrevitLevel)
+            {
+                case 2:  return 10;  // Debug   → logging.DEBUG
+                case 1:  return 20;  // Verbose → logging.INFO
+                default: return 30;  // Quiet   → logging.WARNING (default)
+            }
```
[the remark this answers is not in the document]
Comment (tay0thman) [reply dev/pyRevitLoader/pyRevitAssemblyBuilder/SessionManager/EnvDictionarySeeder.cs]: ddressed — added EnvDictionarySeederTests.cs with 4 tests covering Quiet→30, Verbose→20, Debug→10, and edge cases (negative/out-of-range values default to 30). Method visibility changed from private to internal with InternalsVisibleTo for direct test access.

**Comment by tay0thman** [comment]
Added new commit. https://github.com/pyrevitlabs/pyRevit/pull/3207/commits/64611f2e7f543bdf10ea244ff2a9f0293e602a62

@jmcouffin  another run by copilot is appreciated.

File: dev/pyRevitLoader/pyRevitAssemblyBuilder/SessionManager/EnvDictionarySeeder.cs [hunk 2]
```diff
@@ -70,7 +70,11 @@ public static void Seed(UIApplication uiApp, Assembly runtimeAssembly, string py
                 [KeyIPYVersion]         = ReadIPYVersion(pyRevitRoot),
                 [KeyCPYVersion]         = "3.12.3",    // Known default for the bundled CPython engine
 
-                [KeyLoggingLevel]       = config.LoggingLevel,
+                // Fix for #3203: PyRevitConfig.LoggingLevel returns a pyRevit enum
+                // (0=Quiet, 1=Verbose, 2=Debug) but the Python logger reads this
+                // env var as a Python logging module level (10=DEBUG, 20=INFO, 30=WARNING).
+                // Translate to avoid corrupting the Python logging threshold.
+                [KeyLoggingLevel]       = ToPythonLoggingLevel(config.LoggingLevel),
                 [KeyFileLogging]        = config.FileLogging,
```
[the remark this answers is not in the document]
Comment (tay0thman) [reply dev/pyRevitLoader/pyRevitAssemblyBuilder/SessionManager/EnvDictionarySeeder.cs:78]: Binaries has been pushed again under a new Commit
https://github.com/pyrevitlabs/pyRevit/pull/3207/changes/aae432de2ad18f2b3aab7ec5de4e8df1ebcc2354
